sklearn/utils/tests/test_validation.py

```diff
@@ -835,3 +836,67 @@ def test_check_scalar_invalid(x, target_name, target_type, min_val, max_val,
                      min_val=min_val, max_val=max_val)
     assert str(raised_error.value) == str(err_msg)
     assert type(raised_error.value) == type(err_msg)
+
+
+_psd_cases_valid = {
+    'nominal': ((1, 2), np.array([1, 2]), None, ""),
+    'insignificant_imag': ((5, 5e-5j), np.array([5, 0]), None, ""),
+    'significant neg': ((5, -1), np.array([5, 0]), PSDSpectrumWarning,
+                        "There are significant negative eigenvalues"),
+    'insignificant neg': ((5, -5e-5), np.array([5, 0]), None, "")
+}
+
+
+@pytest.mark.parametrize("lambdas, expected_lambdas, w_type, w_msg",
+                         list(_psd_cases_valid.values()),
+                         ids=list(_psd_cases_valid.keys()))
+@pytest.mark.parametrize("warn_on_zeros", [True, False])
+def test_check_psd_eigenvalues_valid(lambdas, expected_lambdas, w_type, w_msg,
+                                     warn_on_zeros):
+    """Test that check_psd_eigenvalues returns the right output for valid
+    input, possibly raising the right warning"""
+
+    with pytest.warns(w_type, match=w_msg) as w:
```

Comment:
I like the idea of parametrizing this. Nice :)

Reply:
:) I got quite involved in writing [pytest plugins](https://github.com/smarie/ALL_OF_THE_ABOVE#tests) to support data science benchmarks creation, so I can't help but using parametrization now :)